### DISCUSSION TITLE
Python Frontend for Mesa

### DISCUSSION THREAD

**Discussion by ankitk50** [mesa/mesa#1765] [General]
Hi all,

I would start making changes to integrate Solara with Mesa (see #1622 ). The goal would be to have a pure python frontend for Mesa release 3.0. The proposed timeline for the changes is from mid Aug 23 to mid January 24.  

For the start I would work on mesa-examples and would try to come up with a generic code structure we can bring up in Mesa next release.

In coming weeks I would post the progress/discourse on this thread. If you have any suggestions feel free to get in touch.

**Comment by rht** [comment]
It's already in this repo: https://github.com/projectmesa/mesa/blob/main/mesa/experimental/jupyter_viz.py. Your first objective should be to improve the performance of the Matplotlib redraw for larger grid size.

**Comment by Corvince** [comment]
That's awesome! I am excited to see where you go with solara! Lots of opportunities there I think. 

I have also been playing around a few hours with the solara frontend and hope to contribute as well. I was a react developer in my previous job, so let's see if that helps.

**Comment by ankitk50** [comment]
I have started with porting some examples to solara, to gain a better understanding of the framework and to get some more ideas on how to improve the plotting of the grids.

**Reply by tpike3** [reply]
Excellent, also please take a look at #1775 so you can see a related pull request.

**Reply by ankitk50** [reply]
Thanks I will also include the step-count in my examples.

**Reply by rht** [reply]
Can you make sure to open 1 PR for each examples, instead of 1 giant PR? We prefer smaller PRs for faster turnaround rate.

**Comment by tpike3** [comment]
Updated timeline: 

Based on discussions this is the updated timeline I am proposing: 

12 August - Python Frontend Framework Selected (done) 
14 October  - Basic skeleton of front end created
11 November - Working prototype with desired features (1st payment)
9 December - Develop test, refine and ensure code is complaint with contributor standards
13 January - Update tutorials and education support
10 February - Final clean up and release new Mesa version (final payment) 

@ankitk50 Let me know if this works?

**Reply by ankitk50** [reply]
Yes , perfect ! Thanks for accommodating the request!
